community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -472,6 +474,98 @@ def update_topology(lkp: util.Lookup) -> None:
         log.debug("Topology configuration updated. Reconfiguring Slurm.")
         util.scontrol_reconfigure(lkp)
 
+
+def delete_reservations(reservation_map: Dict[str, str]) -> None:
+    slurm_delete_reservation = "scontrol delete reservation {}"
+    for reservation_name in list(reservation_map.keys()):
+      util.run(slurm_delete_reservation.format(reservation_name), timeout=30)
+
+
+def update_slurm_reservation_maintenance(node: str, starttime: str,
+                                         reservation_map: Dict[str, Tuple[str, str]]) -> None:
+    slurm_create_reservation = "scontrol create reservation user=root starttime={} duration=180 nodes={} reservationname={}"
+    slurm_delete_reservation = "scontrol delete reservation {}"
+
+    reservation_name = node + "_maintenance"
+    ## If reservation exists for maintenance, update reservation if time window
+    ## for scheduled maintenance changes.
+    if reservation_name in list(reservation_map.keys()):
+      nodes, maintenance_start_time = reservation_map[reservation_name]
+      del reservation_map[reservation_name]  # remove the reservation from the map.
+
+      if maintenance_start_time == starttime:
+        return
+
+      util.run(slurm_delete_reservation.format(reservation_name), timeout=30)
+      util.run(slurm_create_reservation.format(starttime, nodes, reservation_name), timeout=30)
+
+    else: # Reservation doesn't exist, make reservation during maintenance window.
+      util.run(slurm_create_reservation.format(starttime, node, reservation_name), timeout=30)
+
+
+def get_slurm_reservation_maintenance() -> Dict[str, Tuple[str, str]]:
+    slurm_show_reservation = "scontrol show reservation --json"
+    res = util.run(slurm_show_reservation, timeout=30)
+    all_reservations = json.loads(res.stdout)
+    reservation_map = {}
+
+    for reservation in all_reservations['reservations']:
+        reservation_name = reservation['name']
+        nodes = reservation['node_list']
```

Comment:
Looks like there is an assumption that "our" reservations are always for a single, node.
Please add check `reservation["node_count"] == 1`.
There is a still a possibility to get an ugly name such as `node-[0]`, but it shouldn't happen AFAIU.

Reply:
done.